dir.go

```diff
@@ -211,7 +208,7 @@ func (d *Directory) Uncache(name string) {
 // childFromDag searches through this directories dag node for a child link
 // with the given name
 func (d *Directory) childFromDag(name string) (ipld.Node, error) {
-	return d.unixfsDir.Find(d.ctx, name)
+	return d.unixfsDir.Find(context.Background(), name)
```

Comment:
Yeah, in general, never use `context.Background()` unless something really shouldn't be cancelable. This here will make canceling reads impossible.